src/components/TonConnectUIProvider.vue

```diff
@@ -1,8 +1,9 @@
 <script lang="ts">
```

Comment:
Wait, provider is implemented incorrect. I see, why.

In the issue I put next example of code:

```
// <TonConnectUIProvider />
const tonConnectUI = shallowRef<TonConnectUI | null>(null);

provide('ton-connect-ui', tonConnectUI);

// Injection consumer, for example `useTonWallet`
const tonConnectUi = inject('ton-connect-ui'); // That line of code should be inside `useTonConnectUI`, put it here for simplification
const unsub = shallowRef<() => void>(() => {});
const wallet = shallowRef<...>(null);

onMounted(() => {
  watch(tonConnectUI, (actualTonConnectUI) => {
    unsub(); // Unsubscribe on watcher trigger <-> when tonConnectUI instance is updated (as done in React)

    wallet.value = actualTonConnectUI.wallet;

    unsub.value = actualTonConnectUI.onStatusChange((value) => {
      wallet.value = value;
    });
  });
});

onUnmounted(() => {
  unsub.value();
});
```

Pay attention to the comment `// Injection consumer, for example `useTonWallet``. So all next code should be placed into `useTonWallet`.